pkg/virt-handler/vm.go

```diff
@@ -1572,33 +1582,22 @@ func (d *VirtualMachineController) heartBeat(interval time.Duration, stopCh chan
 			// Label the node if cpu manager is running on it
 			// This is a temporary workaround until k8s bug #66525 is resolved
 			if d.clusterConfig.CPUManagerEnabled() {
-				d.updateNodeCpuManagerLabel()
+				d.updateNodeCpuManagerLabel(cpuManagerPath)
 			}
 		}, interval, 1.2, true, stopCh)
 	}
 }
 
-func (d *VirtualMachineController) updateNodeCpuManagerLabel() {
-	entries, err := filepath.Glob("/proc/*/cmdline")
+func (d *VirtualMachineController) updateNodeCpuManagerLabel(cpuManagerPath string) {
+	content, err := ioutil.ReadFile(cpuManagerPath)
 	if err != nil {
 		log.DefaultLogger().Reason(err).Errorf("failed to set a cpu manager label on host %s", d.host)
 		return
 	}
 
 	isEnabled := false
-	for _, entry := range entries {
-		content, err := ioutil.ReadFile(entry)
-		if os.IsNotExist(err) {
-			// processes can disappear anytime, it is ok if they don't exist anymore
-			continue
-		} else if err != nil {
-			log.DefaultLogger().Reason(err).Errorf("failed to set a cpu manager label on host %s", d.host)
-			return
-		}
-		if strings.Contains(string(content), "kubelet") && strings.Contains(string(content), "cpu-manager-policy=static") {
-			isEnabled = true
-			break
-		}
+	if strings.Contains(string(content), "\"policyName\":\"static\"") {
```

Comment:
Do you think you can parse the file and extract the entry from a parsed json (can be anonymous into maps/arrays)?

Reply:
Yes, it should be possible. I will change.

Reply:
done